contracts/drafts/dao/DAO.sol

```diff
@@ -0,0 +1,117 @@
+pragma solidity ^0.5.10;
+
+import "@openzeppelin/contracts/token/ERC20/ERC20Mintable.sol";
+import "../../issuance/IssuanceEth.sol";
+import "../token/ERC20MultiDividendable.sol";
+
+
+contract DAO is ERC20Mintable, ERC20MultiDividendable, IssuanceEth {
+
+    mapping(address => uint256) public begAmount;
+    uint256 totalAmounts;
+    mapping(address =>
+        mapping(address => uint256)
+    ) public votesForIdeaByHolder;
+    mapping(address => uint256) public totalVotesForIdea;
+    mapping(address => uint256) public totalVotesByHolder;
+    mapping(address => address[]) public backersForIdea;
+    event Here();
+
+    constructor()
+    ERC20Mintable()
+    ERC20MultiDividendable()
+    IssuanceEth(address(this))
+    public
+    {
+        _createState("NEVER");
+        _createTransition("LIVE", "SETUP");
+        _createTransition("FAILED", "SETUP");
+        addMinter(address(this));
+    }
+
+    function transferFunds(address payable _wallet) public onlyOwner {
+        require(currentState == "NEVER", "You can never do this.");
+    }
+
+    function restartFundingRound() public onlyOwner {
+        require(
+            currentState == "LIVE" || currentState == "FAILED",
+            "Initial funding round not ended."
+        );
+        _transition("SETUP");
+    }
+
+    function begMoneyForIdea(uint256 amount, address payable idea) public {
+        require(currentState == "LIVE", "Founders not defined yet.");
+        require(
+            totalAmounts + amount <= address(this).balance,
```

Comment:
Can overflow happens if
totalAmounts = 15
amount = 2^256 - 10
address(this).balance = 10

and `begAmount[idea]` be 2^256 - 10 ?

Reply:
God point! I'll use `SafeMath`, that should do, shouldn't it?